feat: reva app auth (#9079)
owncloud/ocis: services/auth-app/README.md

# Auth-App

The auth-app service provides authentication for 3rd party apps.

## The `auth` Service Family

ocis uses serveral authentication services for different use cases. All services that start with `auth-` are part of the authentication service family. Each member authenticates requests with different scopes. As of now, these services exist:
  -   `auth-basic` handles basic authentication
  -   `auth-bearer` handles oidc authentication
  -   `auth-machine` handles interservice authentication when a user is impersonated
  -   `auth-service` handles interservice authentication when using service accounts

>   -   `auth-app` handles authentication of external 3rd party apps

Reviewer: Note, you need to provide the new service also to the readme's of the other services...
They _should_ / _must_ have a identical list...

Reviewer 2: That's what I did. See `Readme.md` files in `auth-bearer`, `auth-machine` and `auth-service`

> ## Optional Service

Reviewer: ```suggestion
## Service Startup
```

> This service is an optional service that will not run with default settings. To start use it, two envvars need to be set:

Reviewer: ```suggestion
This service is optional and will not run with default settings. To start using it, two envvars need to be set:

```

> This service is an optional service that will not run with default settings. To start use it, two envvars need to be set:

Reviewer: ```suggestion
Because this service is not started automatically, a manual start needs to be initiated which can be done in several ways. To configure the service usage, an environment variable for the proxy service needs to be set to allow app authentication.
```

```bash

> OCIS_ADD_RUN_SERVICES=auth-app # to start the service. Alternatively you can start the service explicitly via the command line.

Reviewer: ```suggestion
OCIS_ADD_RUN_SERVICES=auth-app # to start the service when using the binary deployment. Alternatively you can start the service explicitly via the command line.
```

> OCIS_ADD_RUN_SERVICES=auth-app # to start the service. Alternatively you can start the service explicitly via the command line.
> PROXY_ENABLE_APP_AUTH=true # to allow app authentication. This envvar goes to the proxy service in case of a distributed environment.

Reviewer: ```suggestion
OCIS_ADD_RUN_SERVICES=auth-app  # deployment specific. Add the service to the manual startup list, use with binary deployments. Alternatively you can start the service explicitly via the command line.
PROXY_ENABLE_APP_AUTH=true      # mandatory, allow app authentication. In case of a distributed environment, this envvar needs to be set in the the proxy service.
```

```

## App Tokens

> App Tokens are used to authenticate 3rd party apps. To be able to use an app token, one must first create a token via cli.

Reviewer: ```suggestion
App Tokens are used to authenticate 3rd party access via https like when using curl (apps) to access an API endpoint. These apps need to authenticate themselve as no logged in user authenticates the request. To be able to use an app token, one must first create a token via cli.
```

> PROXY_ENABLE_APP_AUTH=true      # mandatory, allow app authentication. In case of a distributed environment, this envvar needs to be set in the the proxy service.

Reviewer: ```suggestion
PROXY_ENABLE_APP_AUTH=true      # mandatory, allow app authentication. In case of a distributed environment, this envvar needs to be set in the proxy service.
```

```bash
ocis auth-app create --user-name={user-name} --expiration={token-expiration}
```

> Once generated, these tokens can be used to authenticate requests to the oCIS services. They can be passed in any request as `Basic Auth` header.

Reviewer: ```suggestion
Once generated, these tokens can be used to authenticate requests to ocis. They are passed as part of the request as `Basic Auth` header.
```
